bubblegum/xrf/model/guessparam.py

```diff
@@ -0,0 +1,467 @@
+
+    def create_full_param(self,
+                          peak_std=0.07, peak_height=500.0):
+        """
+        Extend the param to full param dict with detailed elements
+        information, and assign initial values from pre fit.
+
+        Parameters
+        ----------
+        peak_std : float
+            approximated std for element peak.
+        peak_height : float
+            initial value for element peak height
+        """
+
+        PC = ParamController(self.param_d)
+        PC.create_full_param()
+        self.param_new = PC.new_parameter
+        factor_to_area = np.sqrt(2*np.pi)*peak_std*0.5
+
+        if len(self.result_dict):
+            for e in self.e_list.split(','):
+                e = e.strip(' ')
+                zname = e.split('_')[0]
+                for k, v in six.iteritems(self.param_new):
+                    if zname in k and 'area' in k:
+                        if self.result_dict.has_key(e):
+                            v['value'] = self.result_dict[e].maxv*factor_to_area
+                        else:
+                            v['value'] = peak_height*factor_to_area
+            self.param_new['compton_amplitude']['value'] = self.result_dict['compton'].maxv*factor_to_area
+            self.param_new['coherent_sct_amplitude']['value'] = self.result_dict['elastic'].maxv*factor_to_area
+
+    def data_for_plot(self):
+        """
+        Save data in terms of K, L, M lines for plot.
+        """
+        self.total_y = {}
+        self.total_y_l = {}
+        self.total_y_m = {}
+        new_dict = {k: v for (k, v) in six.iteritems(self.result_dict) if v.status}
+        for k, v in six.iteritems(new_dict):
+            if 'K' in k:
+                self.total_y[k] = self.result_dict[k].spectrum
+            elif 'L' in k:
+                self.total_y_l[k] = self.result_dict[k].spectrum
+            elif 'M' in k:
+                self.total_y_m[k] = self.result_dict[k].spectrum
+            else:
+                self.total_y[k] = self.result_dict[k].spectrum
+
+    def save(self, fname='param_default1.json'):
+        """
+        Save full param dict into a file at result directory.
+        The name of the file is predefined.
+
+        Parameters
+        ----------
+        fname : str, optional
+            file name to save updated parameters
+        """
+        self.save_elist()
+        self.create_full_param()
+        fpath = os.path.join(self.result_folder, fname)
+        with open(fpath, 'w') as outfile:
+            json.dump(self.param_new, outfile,
+                      sort_keys=True, indent=4)
+
+    def save_as(self):
+        """
+        Save full param dict into a file.
+        """
+        self.save_elist()
+        self.create_full_param()
+        with open(self.file_path, 'w') as outfile:
+            json.dump(self.param_new, outfile,
+                      sort_keys=True, indent=4)
+
+
+def pre_fit_linear(y0, fitting_parameters):
```

Comment:
The core calculation function is already in skxray. This function just performs as a wrapper.